### ISSUE TITLE
Replace Tiny MCE with something else

### ISSUE THREAD

**Issue by eternoendless** [PrestaShop/PrestaShop#12561]
We need to find a better alternative to Tiny MCE because of unclear license compatibility.

**Comment by rdy4ever** [comment]
May I suggest [Summernote](https://github.com/summernote/summernote/)?

**Comment by dennispw** [comment]
Will Markdown ever be an alternative?

**Comment by rdy4ever** [comment]
@dennispw I really can't see that happening. The Prestashop admin must be non-tech-friendly, as one cannot ask from regular shop owners to know how to use markdown. I guess markdown could be added as optional somehow (like another button next to view source - "use markdown"), but never as the default editor.

**Comment by rdy4ever** [comment]
@dennispw [this](https://github.com/nhnent/tui.editor) looks promissing. A hybrid WYSIWYG/Markdown editor. I personally wouldn't mind seeing it in PS.

**Comment by sarjon** [comment]
https://ckeditor.com/, it integrates well with Symfony using FOSCKEditorBundle. Although im not  sure about license.

**Comment by dennispw** [comment]
I completely understand the need for a WYSIWYG. On smaller personal projects (non PS related) I have however had no issues with people getting used to Markdown after explaining the benefits and how it works. But then again, I can definately see why such an implementation could cause a stir. I was merely asking as I always prefer Markdown whenever possible.

**Comment by davidglezz** [comment]
Hard task: It is difficult to evaluate which one includes better features with a compatible license.
- TinyMCE
- Ckeditor5
- [Quill](https://github.com/quilljs/quill)
- [ProseMirror](https://github.com/ProseMirror/prosemirror)
- More https://github.com/JefMari/awesome-wysiwyg

My wishes
- I wish It had a good HTML editor (Monaco)
- Possibility of applying predefined classes to elements from a menu (.alert, .table, .btn, ...)
- Html cleaning

 TinyMCE already does last 2.

TinyMCE is a little bugged sometimes

![tmce](https://user-images.githubusercontent.com/194784/75854961-8f3f1900-5df1-11ea-9735-68324b4aa947.gif)

**Comment by eternoendless** [comment]
The way I see it, the editor should be a module that hooks into a simple textarea field. This way people would be able to switch editors as they see fit.

**Comment by Oksydan** [comment]
Yes but module developers might face problem with text editor compatibility this way. 
Recently I was creating module for advanced product description, where you could create some kind of grid.
I had to reinitialize tinymce after ajax calls with some options to make tinymce do what I pleases.
If editor API if unknown for developer, we have to create some kind of facet so module developers can just initialize/reinitialize it easily but problem of editor options still remains 😕.

**Comment by darkf3d3** [comment]
Has the possibility of replacing TinyMCE been discussed anymore?

**Comment by boo-code** [comment]
Before picking a replacement, the licence premise is worth re-checking: TinyMCE has been relicensed twice since this was opened, and one of those versions is MIT.

Taken from the vendor's own licence file at each tag:

| TinyMCE | Licence file | Licence |
| --- | --- | --- |
| 4.9.11 | `LICENSE.TXT` | LGPL 2.1 |
| 5.10.9 | `LICENSE.TXT` | LGPL 2.1 |
| 6.0.0, 6.8.5 | `LICENSE.TXT` | MIT |
| 7.0.0, 8.0.0 | `LICENSE.md` | GPL v2 or later |

Core vendors 4.9.11 (`js/tiny_mce/tinymce.min.js`, header `// 4.9.11 (2020-07-13)`), and `docs/licenses/tinymce-smarty-swift.txt` carries the LGPL 2.1 text for it. 4.9.11 is also the last 4.x tag upstream.

So the GPL relicence is 7.x, not what is shipped, and 6.x is MIT. For comparison with an alternative named above, CKEditor 5 is dual GPL v2 or later / commercial (`packages/ckeditor5/LICENSE.md`). If the licence is the driver for this issue, upgrading 4.9.11 to 6.8.5 answers it without changing editor.

Scope of that upgrade against the current tree: 18 files mention the editor, but the integration surface is five - `js/admin/tinymce.inc.js`, `js/admin/tinymce_loader.js`, `js/admin/products.js`, `admin-dev/themes/new-theme/js/components/tinymce-editor.js`, `admin-dev/themes/default/js/bundle/product/form.js`. The part that does not port mechanically is the vendored custom skin `js/tiny_mce/skins/prestashop/` (v4 format: `skin.min.css`, `content.min.css`, `content.inline.min.css`, `fonts/`), selected by name in two of those files as `skin: 'prestashop'`.
